### ISSUE TITLE
Investigate transaction type handling on dune.com

### ISSUE THREAD

**Issue by arthurgousset** [celo-org/celo-monorepo#10424]
@karlb shared that tx type handling on dune looks wrong to me. I'm pretty sure we don't have only legacy tx as shown in this query: https://dune.com/queries/2706785 (source: [slack](https://celo-org.slack.com/archives/C05AA4A0A3V/p1688979266797019)). 

@carterqw2 shared that when he last ran a query against cLabs' internal db, he found very few non-legacy type transactions giving the results in Karl's query some credit (source: [slack](https://celo-org.slack.com/archives/C01V5UXJ6RY/p1686837027868999?thread_ts=1686758960.621619&cid=C01V5UXJ6RY))

Here 0 is a legacy tx, null is 0 for cases when Blockscout (our indexer) didn’t have a notion of types yet, and 1 presumably non-legacy transactions.

| Type |   Count   |
|--|--|
| null | 103618732 |
|   0  | 150617804 |	
|   1  |    947    |

Ask: 

+ [ ] assert that dune is correctly indexing events and not omitting non-legacy transactions (even if there are very few)

**Comment by yerdua** [comment]
Rough initial findings:

I confirmed that in dune's dataset, all transactions have `type = Legacy`. 

To check whether non-legacy transactions are missing or mis-typed, I identified a couple specific transactions to spot-check. Via BigQuery, I looked up the hashes of arbitrary `type = 1` and `type = 2` transactions. A query in dune correctly found these transactions by their hashes, but showed `Legacy` as the `type` for both of them.

So, this suggests that dune is not omitting non-legacy transactions. Instead, they're getting the wrong type.


The next thing to investigate would be how and where the translation of an integer into a string is happening. In other words, how does `null, 0, 1, 2` become `"Legacy"`?
